README.md

```diff
@@ -27,37 +32,51 @@ Suppose you have the source code of smart contract in the folder `contract` unde
+Run `pchain_compile` with the arguments **source** and **destination** to specify the folder of the source code and the folder for saving the result.
 
-On a Linux Bash Shell:
-      
 ```sh
-$ ./pchain_compile build --source /home/user/contract --destination /home/user/result
+pchain_compile build --source /home/user/contract --destination /home/user/result
+```
+
+Once complete, the console displays message:
+
+```text
 Build process started. This could take several minutes for large contracts.
 
 Finished compiling. ParallelChain Mainnet smart contract(s) ["contract.wasm"] are saved at (/home/user/result).
 ```
 
-On a Windows Shell:
+Your WebAssembly smart contract is now saved with file extension `.wasm` at the destination folder. 
+
+If you are running on Windows, here is the example output:
 ```powershell
 $ .\pchain_compile.exe build --source 'C:\Users\user\contract' --destination 'C:\Users\user\result'
 Build process started. This could take several minutes for large contracts.
 
 Finished compiling. ParallelChain Mainnet smart contract(s) ["contract.wasm"] are saved at (C:\Users\user\result).
 ```
 
-Explanation about the command and its arguments can be displayed by appending "help" or "--help" to `pchain_compile`.
+To understand more about the commands and arguments, run `pchain_compile build --help`.
 
-## Toolchain
+## About Docker Image
 
-`pchain_compile` utilizes a docker_image hosted on a public DockerHub repository of ParallelChain see [here](https://hub.docker.com/r/parallelchainlab/pchain_compile) for the build process. Required components include:
+`pchain_compile` pulls docker image from ParallelChain Lab's official DockerHub [repository](https://hub.docker.com/r/parallelchainlab/pchain_compile) for the build process. The docker image provides an environment with installed components:
 - rustc: compiler for Rust.
 - wasm-snip: WASM utility which removes functions that are never called at runtime.
 - wasm-opt: WASM utility to load WebAssembly in text format and run Binaryen IR passes to optimize its size. For more information on Binaryen IR see [here](http://webassembly.github.io/binaryen/).
 
-The docker images utilize a toolchain whose versions of each component are shown in the following table:
+There are different tags of the docker image. They vary on the versions of the components. The table below describes the tags and their differences.
 
 |Image Tag |rustc |wasm-snip |wasm-opt |
 |:---|:---|:---|:---|
-|0.4.2 |1.71.0 |0.4.0 |114 |
-|mainnet01 |1.66.1 |0.4.0 |109 |
+|0.4.3 | 1.77.1 | 0.4.0| 114|
+|0.4.2 | 1.71.0 | 0.4.0 | 114 |
+|mainnet01 | 1.66.1 | 0.4.0 | 109 |
+
+To build smart contract in a specific docker environment, run with argument **use-docker-tag**. For example,
+
+```sh
+pchain_compile build --source /home/user/contract --destination /home/user/result --use-docker-tag 0.4.3
+```
+
+If **use-docker-tag** is not used, the tag being used is with the same version as `pchain_compile`.
```

Comment:
Does it mean it uses the default version set by `pchain_compile`?

Reply:
It means the version of the program determines the docker image tag. For example, `pchain_compile` v0.4.3 will pull the docker image with tag `0.4.3` if **use-docker-tag** is not set.